easybuild/easyblocks/n/neuron.py

```diff
@@ -60,85 +62,135 @@ def extra_options():
+                run_cmd(cmd, simple=True, log_all=True, log_ok=True)
 
-            try:
-                os.chdir(pwd)
-            except OSError as err:
-                raise EasyBuildError("Failed to change back to %s: %s", pwd, err)
+                try:
+                    os.chdir(pwd)
+                except OSError as err:
+                    raise EasyBuildError("Failed to change back to %s: %s", pwd, err)
 
     def sanity_check_step(self):
         """Custom sanity check for NEURON."""
         shlib_ext = get_shared_lib_ext()
         binpath = os.path.join(self.hostcpu, 'bin')
         libpath = os.path.join(self.hostcpu, 'lib', 'lib%s.' + shlib_ext)
-        binaries = ["bbswork.sh", "hel2mos1.sh", "ivoc", "memacs", "mkthreadsafe", "modlunit", "mos2nrn",
-                    "mos2nrn2.sh", "neurondemo", "nocmodl", "oc"]
-        binaries += ["nrn%s" % x for x in ["gui", "iv", "iv_makefile", "ivmodl", "mech_makefile", "oc", "oc_makefile",
-                                           "ocmodl"]]
-        libs = ["ivoc", "ivos", "memacs", "meschach", "neuron_gnu", "nrniv", "nrnmpi", "nrnoc", "nrnpython", "oc",
-                "ocxt", "scopmath", "sparse13", "sundials"]
-
         # hoc_ed is not included in the sources of 7.4. However, it is included in the binary distribution.
         # Nevertheless, the binary has a date old enough (June 2014, instead of November 2015 like all the
         # others) to be considered a mistake in the distribution
+        binaries = []
         if LooseVersion(self.version) < LooseVersion('7.4'):
             binaries.append("hoc_ed")
 
-        custom_paths = {
+        sanity_check_dirs = ['share/nrn']
+        if LooseVersion(self.version) < LooseVersion('7.8.1'):
+            binaries += ["bbswork.sh", "hel2mos1.sh", "ivoc", "memacs", "mkthreadsafe", "modlunit", "mos2nrn",
+                         "mos2nrn2.sh", "neurondemo", "nocmodl", "oc"]
+            binaries += ["nrn%s" % x for x in ["gui", "iv", "iv_makefile", "ivmodl", "mech_makefile", "oc",
+                                               "oc_makefile", "ocmodl"]]
+            libs = ["ivoc", "ivos", "memacs", "meschach", "neuron_gnu", "nrniv", "nrnmpi", "nrnoc", "nrnpython",
+                    "oc", "ocxt", "scopmath", "sparse13", "sundials"]
+            sanity_check_dirs += ['include/nrn']
+        # list of included binaries changed with cmake. See
+        # https://github.com/neuronsimulator/nrn/issues/899
+        else:
+            binaries += ["mkthreadsafe", "modlunit", "neurondemo", "nocmodl", "nrngui", "nrniv", "nrnivmodl",
+                         "nrnmech_makefile", "nrnpyenv.sh", "set_nrnpyenv.sh", "sortspike"]
+            libs = ["nrniv", "rxdmath"]
+            sanity_check_dirs += ['include']
+            if self.with_python:
+                sanity_check_dirs += [os.path.join("lib", "python"),
+                                      os.path.join("lib", "python%(pyshortver)s", "site-packages")]
```

Comment:
You should use `self.pylibdir` here, then you don't need the `%(pyshortver)s` template anymore?

Reply:
`self.pylibdir`, or `det_pylibdir()`  both results in only the last version of python being tested.

Reply:
with the `sanity_check_dirs`, it does check the 3 versions : 
```
== 2021-01-07 19:32:17,898 build_log.py:266 INFO sanity checking...
== 2021-01-07 19:32:17,899 easyblock.py:3057 INFO Starting sanitycheck step
== 2021-01-07 19:32:17,899 easyblock.py:3063 INFO Running method sanity_check_step part of step sanitycheck
== 2021-01-07 19:32:17,900 easyblock.py:2427 INFO Running sanity check with extra modules: python/3.6
== 2021-01-07 19:32:17,900 easyblock.py:2531 INFO Using (only) sanity check paths specified by easyconfig file: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['include', 'share/nrn', 'lib/python', 'lib/python3.6/site-packages']}
== 2021-01-07 19:32:31,884 easyblock.py:2756 INFO Loading extra modules for sanity check: python/3.6
== 2021-01-07 19:32:33,385 easyblock.py:2427 INFO Running sanity check with extra modules: python/3.7
== 2021-01-07 19:32:33,385 easyblock.py:2531 INFO Using (only) sanity check paths specified by easyconfig file: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['include', 'share/nrn', 'lib/python', 'lib/python3.7/site-packages']}
== 2021-01-07 19:32:41,735 easyblock.py:2756 INFO Loading extra modules for sanity check: python/3.7
== 2021-01-07 19:32:43,209 easyblock.py:2427 INFO Running sanity check with extra modules: python/3.8
== 2021-01-07 19:32:43,209 easyblock.py:2531 INFO Using (only) sanity check paths specified by easyconfig file: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['include', 'share/nrn', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-07 19:32:52,195 easyblock.py:2756 INFO Loading extra modules for sanity check: python/3.8
``` 

whereas with either `self.pylibdir` or `det_pylibdir()` in the `custom_paths`, then we get : 
```
[mboisson@build-node2 n]$ grep "Using customized" /home/mboisson/.local/easybuild/software/2020/avx2/MPI/intel2020/openmpi4/neuron/7.8.2/easybuild/easybuild-NEURON-7.8.2-20210108.160549.log
== 2021-01-08 16:04:44,144 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-08 16:04:59,627 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-08 16:05:11,236 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/mkthreadsafe', 'bin/modlunit', 'bin/neurondemo', 'bin/nocmodl', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nrnmech_makefile', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
[mboisson@build-node2 n]$ grep "Using customized" /home/mboisson/.local/easybuild/software/2020/avx2/MPI/intel2020/openmpi4/neuron/7.8.2/easybuild/easybuild-NEURON-7.8.2-20210108.161444.log
== 2021-01-08 16:13:39,381 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/neurondemo', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nocmodl', 'bin/modlunit', 'bin/nrnmech_makefile', 'bin/mkthreadsafe', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-08 16:13:55,305 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/neurondemo', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nocmodl', 'bin/modlunit', 'bin/nrnmech_makefile', 'bin/mkthreadsafe', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
== 2021-01-08 16:14:06,847 easyblock.py:2537 INFO Using customized sanity check paths: {'files': ['bin/neurondemo', 'bin/nrngui', 'bin/nrniv', 'bin/nrnivmodl', 'bin/nocmodl', 'bin/modlunit', 'bin/nrnmech_makefile', 'bin/mkthreadsafe', 'bin/nrnpyenv.sh', 'bin/set_nrnpyenv.sh', 'bin/sortspike', 'lib/libnrniv.so', 'lib/librxdmath.so'], 'dirs': ['share/nrn', 'include', 'lib/python', 'lib/python3.8/site-packages']}
```